src/test/java/org/opensearch/ad/transport/ADJobRunnerTransportActionTests.java

```diff
@@ -0,0 +1,93 @@
+package org.opensearch.ad.transport;
+
+import static org.mockito.Mockito.mock;
+
+import java.io.IOException;
+import java.time.Instant;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.ad.util.RestHandlerUtils;
+import org.opensearch.client.Client;
+import org.opensearch.extensions.action.ExtensionActionRequest;
+import org.opensearch.extensions.action.ExtensionActionResponse;
+import org.opensearch.jobscheduler.spi.JobDocVersion;
+import org.opensearch.jobscheduler.spi.JobExecutionContext;
+import org.opensearch.jobscheduler.spi.utils.LockService;
+import org.opensearch.jobscheduler.transport.ExtensionJobActionRequest;
+import org.opensearch.jobscheduler.transport.JobRunnerRequest;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.tasks.Task;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.TransportService;
+
+public class ADJobRunnerTransportActionTests extends OpenSearchIntegTestCase {
+
+    private ADJobRunnerTransportAction action;
+
+    private Task task;
+
+    private ActionListener<ExtensionActionResponse> response;
+
+    private ExtensionActionRequest extensionActionRequest;
+
+    private LockService lockService;
+
+    private SDKRestClient sdkRestClient;
+
+    private JobExecutionContext jobExecutionContext;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        sdkRestClient = mock(SDKRestClient.class);
+        action = new ADJobRunnerTransportAction(mock(TransportService.class), mock(ActionFilters.class), sdkRestClient);
+
+        task = mock(Task.class);
+        lockService = new LockService(mock(Client.class), clusterService());
```

Comment:
Why are we mocking `Client.class` here when we are using `SDKRestClient`?

Reply:
This is a mock for LockService. For SDKRestClient in LockService is not in the scope of this PR. Moreover, later when @dbwiddis will fix SDKRestClient and dependency injection in the whole project I will fix this.